### PR TITLE
Add k8s.volume.{name,type} attributes

### PR DESCRIPTION
**Pull request by ChrsMark**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/1164

## Changes

Please provide a brief description of the changes here.

Note: if the PR is touching an area that is not listed in the [existing areas](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md), or the area does not have sufficient [domain experts coverage](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/CODEOWNERS), the PR might be tagged as [experts needed](https://github.com/open-telemetry/semantic-conventions/labels/experts%20needed) and move slowly until experts are identified.

Add k8s.volume.{name,type} attributes.

From https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.102.0/receiver/kubeletstatsreceiver/internal/kubelet/conventions.go#L7

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/resource/k8s.yaml [hunk 1]
```diff
@@ -100,3 +100,12 @@ groups:
     attributes:
       - ref: k8s.cronjob.uid
       - ref: k8s.cronjob.name
+
+  - id: k8s.volume
```
Comment (jsuereth) [inline model/resource/k8s.yaml:104]: Let's avoid adding anything to resource until Entities SiG makes more progress.

I'd love to do more here, but right now Identifying + Descriptive attributes are blended in Resource and this is problematic for our long term success.

Let's remove the resource part, but we can reserve the other attributes with the meaning you suggest for general usage in OTEL.

**Review by jsuereth — CHANGES_REQUESTED** [review]

File: model/resource/k8s.yaml (hunk 1 above)
Comment (TylerHelmuth) [reply model/resource/k8s.yaml:104]: @jsuereth does this mean the semantic conventions are not accepting any more resource attribute proposals until Entities is stabilized?

Comment (TylerHelmuth) [reply model/resource/k8s.yaml]: @jsuereth based on the discussion around resource attributes during today's SIG meeting, I believe at least `k8s.volume.name` meets the criteria of "not changing over the lifetime of the object".  I believe that `k8s.volume.type` also meets the criteria, but @ChrsMark can you confirm that a volume's type cant change without recreating?

Comment (ChrsMark) [reply model/resource/k8s.yaml]: From https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement:

> Pod updates may not change fields other than spec.containers[*].image, spec.initContainers[*].image, spec.activeDeadlineSeconds or spec.tolerations. For spec.tolerations, you can only add new entries.

However thinking of it again, my concern is if we should make these resource attributes or metric attributes.
In `kubeletstats` we are going to use them as metric attributes right (for the `k8s.volume.*` metrics) ?

It seems we are in need of metric attributes here. So it sounds correct if we only keep them in the registry for now with a purpose similar to the [`system.device`](https://github.com/open-telemetry/semantic-conventions/blob/v1.26.0/docs/attributes-registry/system.md?plain=1#L24).

**Comment by ChrsMark** [comment]
@open-telemetry/semconv-k8s-approvers could you take a look?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by ChrsMark** [comment]
Thank's @lmolkova for the comments! Should be addressed now.

**Review by lmolkova — APPROVED** [review]

File: model/registry/k8s.yaml [hunk 2]
```diff
@@ -177,3 +177,48 @@ groups:
...
+              brief: "A [local](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#local) volume"
+              stability: experimental
+            - id: gce_persistent_disk
+              value: 'gcePersistentDisk'
+              brief: "A [gcePersistentDisk](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#gcepersistentdisk) volume"
+              stability: experimental
+            - id: glusterfs
+              value: 'glusterfs'
+              brief: "A [glusterfs](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#glusterfs) volume"
+              stability: experimental
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply model/registry/k8s.yaml]: done: [c511e27](https://github.com/open-telemetry/semantic-conventions/pull/1251/commits/c511e279344c506cb44979d887cc34cc4cab54e2)

File: model/registry/k8s.yaml [hunk 3]
```diff
@@ -177,3 +177,48 @@ groups:
...
+              value: 'secret'
+              brief: "A [secret](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#secret) volume"
+              stability: experimental
+            - id: local
+              value: 'local'
+              brief: "A [local](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#local) volume"
+              stability: experimental
+            - id: gce_persistent_disk
+              value: 'gcePersistentDisk'
+              brief: "A [gcePersistentDisk](https://v1-29.docs.kubernetes.io/docs/concepts/storage/volumes/#gcepersistentdisk) volume"
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply model/registry/k8s.yaml]: done: [c511e27](https://github.com/open-telemetry/semantic-conventions/pull/1251/commits/c511e279344c506cb44979d887cc34cc4cab54e2)

**Comment by ChrsMark** [comment]
Hey @jsuereth, could you take another looks please?

**Review by jsuereth — APPROVED** [review]

File: model/resource/k8s.yaml (hunk 1 above)
Comment (jsuereth) [reply model/resource/k8s.yaml]: See: https://github.com/open-telemetry/oteps/pull/264 for the path forward I'm proposing here.  Once we've agreed to what we're doing, we can unlock moving these attributes into Resource/Entity sections in Semconv.

Comment (ChrsMark) [reply model/resource/k8s.yaml]: Thank's for the head's up!

**Comment by ChrsMark** [comment]
The failure is unrelated. I have filed https://github.com/open-telemetry/semantic-conventions/pull/1332 to fix this.

**Pull request merged**